### PR TITLE
Fix the broken dictionary traversal: actually runs the system tests (❗).

### PR DESCRIPTION
**Pull request by samcunliffe**
### Dependencies

- #119

### CODE REVIEW

**Comment by samcunliffe** [comment]
MacOS test failures as expected.

**Review by willGraham01 — APPROVED** [review]
Looks good - once #119 is merged in the macOS tests should pass and we can merge this in too.

Will approve, pending a test re-run after #119 is merged.

**Comment by willGraham01** [comment]
Merging in main following successful merge of #119, and rerunning MacOS tests.

**Pull request merged**
